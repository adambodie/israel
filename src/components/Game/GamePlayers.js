import React, { Component} from 'react';

export default class GamePlayers extends Component {

	handleChange() {
		this.props.handleClick();
	}
	render() {
		const { result, buttons, playerScore, opponentScore, playerWin, opponentWin, draw } = this.props;
		return (
		<div>
			{this.props.start ? (
				<div className='result'>
					<div className="flex">
						<div>
							<h3>Player</h3>
							<img src={`https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/${buttons[playerScore].name}.png`} alt={buttons[playerScore].name} />
						</div>
						<div>
							<h3>Computer</h3>
							<img src={`https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/${buttons[opponentScore].name}.png`} alt={buttons[opponentScore].name} />
						</div>
					</div>
					<div className="resultText">
						<h4>{result}</h4>
					</div>
					<div className="score">
						<div>
							<h4>Player</h4>
							<h4>{playerWin}</h4>
						</div>
						<div>
							<h4>Draw</h4>
							<h4>{draw}</h4>
						</div>
						<div>
							<h4>Computer</h4>
							<h4>{opponentWin}</h4>
						</div>
					</div>
				</div>
				) : (
				<div className='begin'>
					<h2>Press any button to begin</h2>
				</div>
				)
			}
			</div>
		)
	}
}

