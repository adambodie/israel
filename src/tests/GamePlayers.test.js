import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GamePlayers from '../components/Game/GamePlayers';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<GamePlayers opponentScore={0} 
        playerScore={0} 
        playerWin={0} 
        opponentWin={0} 
        draw={0} 
        start={false} 
        /*handleChange={this.handleClick}
        buttons={this.state.buttons}*/
        result={''}/>, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
	const tree = renderer.create(
      <GamePlayers 
      opponentScore={0} 
      playerScore={0} 
      playerWin={0} 
      opponentWin={0} 
      draw={0} 
      start={false} 
      /*handleChange={this.handleClick}
      buttons={this.state.buttons}*/
      result={''} 
      />
  ).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders a `h2`', () => {
	const wrapper = shallow(<GamePlayers />);
	expect(wrapper.find('h2').length).toEqual(1);
 });

