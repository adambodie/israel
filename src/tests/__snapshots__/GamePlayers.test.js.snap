// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="begin"
  >
    <h2>
      Press any button to begin
    </h2>
  </div>
</div>
`;
